crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -1537,58 +1537,47 @@ impl Analyzer<'_, '_> {
     pub(crate) fn expand_intrinsic_types(&mut self, span: Span, ty: &Intrinsic) -> VResult<Type> {
         let arg = &ty.type_args;
 
-        match ty.kind {
-            IntrinsicKind::Uppercase | IntrinsicKind::Lowercase | IntrinsicKind::Capitalize | IntrinsicKind::Uncapitalize => {
-                match arg.params[0].normalize() {
-                    Type::Lit(LitType { lit: RTsLit::Str(s), .. }) => {
-                        let new_val = match ty.kind {
-                            IntrinsicKind::Uppercase => s.value.to_uppercase(),
-                            IntrinsicKind::Lowercase => s.value.to_lowercase(),
-                            IntrinsicKind::Capitalize => {
-                                if s.value.is_empty() {
-                                    "".into()
-                                } else {
-                                    let mut res = String::new();
-                                    let mut chars = s.value.chars();
-
-                                    res.extend(chars.next().into_iter().flat_map(|v| v.to_uppercase()));
-                                    res.push_str(chars.as_str());
-
-                                    res
-                                }
-                            }
-                            IntrinsicKind::Uncapitalize => {
-                                if s.value.is_empty() {
-                                    "".into()
-                                } else {
-                                    let mut res = String::new();
-                                    let mut chars = s.value.chars();
-
-                                    res.extend(chars.next().into_iter().flat_map(|v| v.to_lowercase()));
-                                    res.push_str(chars.as_str());
-
-                                    res
-                                }
-                            }
-                        };
+        match self.normalize(None, Cow::Borrowed(&arg.params[0]), Default::default())?.as_ref() {
```

Comment:
Please use `.normalize()` instead of `as_ref()`

Reply:
Using `normalize` will fail, as it gets to the branch using `assign_to_intrinsic`.
Is fixing this included in the issue? Or, is it better to leave it as `as_ref()`?

Reply:
I mean `Type.normalize()`